### PR TITLE
Add X-Powered-By header

### PR DESCRIPTION
**Pull request by faustinoaq**
### Description of the Change

Show `X-Powered-By` in response headers.

screenshot_20170913_121131

Similar as Kemal is doing:

kemal

### Alternate Designs

Nop

### Benefits

People can see what sites do use Amber :)

### Possible Drawbacks

Nop

### CODE REVIEW

File: spec/amber/router/context_spec.cr [hunk 1]
```diff
@@ -267,4 +267,14 @@ describe HTTP::Server::Context do
       end
     end
   end
+
+  describe "X-Powered-By Header" do
+    it "initializes context with X-Powered-By: Amber" do
+      headers = HTTP::Headers.new
+      request = HTTP::Request.new("GET", "/", headers)
+      context = create_context(request)
+      context.response.headers.add "X-Powered-By", "Amber"
```
Comment (drujensen) [inline spec/amber/router/context_spec.cr]: In order to test if the pipeline is correctly adding the header, you shouldn't set this in the test.  you should let the pipeline add it.

Comment (faustinoaq) [reply spec/amber/router/context_spec.cr]: Ok, let me fix this

Comment (drujensen) [reply spec/amber/router/context_spec.cr]: Try something like this:
```
pipeline = Amber::Pipe::Pipeline.new
pipeline.call(context)
context.response.headers["X-Powered-By"].should eq "Amber"
```

Comment (eliasjpr) [reply spec/amber/router/context_spec.cr]: @faustinoaq @drujensen is correct you can unit test the Pipeline and verify that the header is set. That's a simple test setup.

**Comment by faustinoaq** [comment]
Travis is failing with latest commit, but it works in my local machine:

Some clue ?

image

**Comment by faustinoaq** [comment]
Hi @drujensen @eliasjpr  I think `Amber::Pipe:::Pipeline.new` is not working well :sweat_smile:

**Comment by faustinoaq** [comment]
Oh, I see my local spec are failing too if I execute all specs

screenshot_20170914_191116

Why if I execute the single test it pass?

ok

**Comment by faustinoaq** [comment]
Travis is happy now, and the pipeline is well tested :)

@drujensen Can we squash and merge this PR now ?

**Comment by drujensen** [comment]
@faustinoaq thanks!

**Pull request merged**